### DISCUSSION TITLE
How can I add custom properties to row and cell?

### DISCUSSION THREAD

**Discussion by gcor** [TanStack/table#2477] [General]
Hello. How can I add custom properties to rows and cells?

For example, I want to get row.serverID and cell.serverID in each cell :

```javascript
{
    Cell: ({ row, value: cell }) => {
        console.log(row.serverID, cell.serverID)
        return cell.value;
    }
}
```

**Comment by agneyaloya** [comment]
1. For values that are unique to each row:
I'm making this happen by sending the data points I need (like 'serverID') as a column to the data but making it a hidden column so the user doesn't see it.

2. For attributes that are common to a column:
I am defining these as key: value pairs in the columns array of the component.

`const columns = React.useMemo(
            () => [
            {
                    Header: 'Name',
                    accessor: 'person_name',
                    // disableFilters: true,
                    width: 70,
                    customAttribute: 'This value is accessible to the Name column',
                },
]`

Used later in the `<table>` as:
`<Tooltip title={column.customAttribute} arrow>`

Hope that helps!
